DashAI/back/api/api_v1/endpoints/explainers.py

```diff
@@ -0,0 +1,253 @@
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.get("/")
+async def get_explainers(
+    run_id: Union[int, None] = None, db: Session = Depends(get_db)
+):
+    """Return the available explainers in the database.
+    The explainers can be filtered by run_id.
+
+    Parameters
+    ----------
+    run_id: Union[int, None], optional
+        If specified, the function will return all the explainers associated with
+        the run, by default None.
+
+    Returns
+    -------
+    List[dict]
+        A list of dict containing explainers.
+
+    Raises
+    ------
+    HTTPException
+        If explainer does not exist in the DB.
+    """
+    try:
+        if run_id is not None:
+            explainers = db.scalars(
+                select(Explainer).where(Explainer.run_id == run_id)
+            ).all()
+            if not explainers:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explainer associated with Run not found",
+                )
+        else:
+            explainers = db.query(Explainer).all()
+
+    except exc.SQLAlchemyError as e:
+        log.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal database error",
+        ) from e
+
+    return explainers
+
+
+@router.get("/{explainer_id}")
+async def get_explainer(explainer_id: int, db: Session = Depends(get_db)):
+    """Return the explainer with id explainer_id from the database.
+
+    Parameters
+    ----------
+    explainer_id : int
+        id of the explainer to query.
+
+    Returns
+    -------
+    dict
+        Dict with the specified explainer id.
+
+    Raises
+    ------
+    HTTPException
+        If the explainer with id explainer_id is not registered in the DB.
+    """
+    try:
+        explainer = db.get(Explainer, explainer_id)
+        if not explainer:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="Explainer not found",
+            )
+
+    except exc.SQLAlchemyError as e:
+        log.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal database error",
+        ) from e
+
+    return explainer
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED)
+async def upload_explainer(
+    params: ExplainerParams,
+    db: Session = Depends(get_db),
+):
+    """Endpoint to create an explainer
+
+    Parameters
+    ----------
+    run_id: int
+        Id of the run associated with the explainer
+    explainer_name: str
+        Name of the explainer
+    parameters: dict
+        Explainer configuration parameters
+
+    Returns
+    -------
+    dict
+        Dict with the new explainer.
+
+    Raises
+    ------
+    HTTPException
+        If the explainer file cannot be saved.
+    """
+    try:
```

Comment:
The whole function should be wrapped in this try expect, because:

1. you save the explainer in the database, which is not currently included in the current exception handler.
2. you save files, which is not considered.

With two different except, one for each case (sqlalchemyerror and oserror)

Reply:
Check `delete_experiment`!